include/aws/common/math.msvc.inl

```diff
@@ -67,8 +83,17 @@ AWS_STATIC_IMPL int aws_mul_u32_checked(uint32_t a, uint32_t b, uint32_t *r) {
  * If a + b overflows, returns AWS_OP_ERR; otherwise adds
  * a + b, returns the result in *r, and returns AWS_OP_SUCCESS.
  */
-AWS_STATIC_IMPL int aws_add_u64_checked(uint32_t a, uint32_t b, uint32_t *r) {
-    if (_addcarry_u64(0, a, b, *r)) {
+AWS_STATIC_IMPL int aws_add_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
+    uint8_t c_in = 0u;
+    if (a == 0) {
+        /* Fallback for any one is zero, as a weird bug happened if a=0, returned carry out will always be 1 at certain
+         * platform (MSVC 15) */
```

Comment:
I have doubts that this is a compiler bug ???

I fed some similar code into Compiler Explorer and I get identical assembly from mvsc [2015](https://gcc.godbolt.org/z/8MMxsTMe3) and [2017](https://gcc.godbolt.org/z/oo7d79eGs)

Reply:
Hmmm, yeah, that's weird... 

But the two CI runs on the same type of image, the only difference is the compiler version. I don't know why...